frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -823,3 +825,74 @@ export const handleRedirect = async (
   const redirectURL = (await perspectiveData.properties.importRedirectURL())(project);
   history.push(redirectURL);
 };
+
+export const getTelemetryImport = (values: GitImportFormData) => {
+  if (!values) return {};
+  // get devfile data if exists
+  let devfileJson;
+  if (values.devfile?.devfileContent) {
+    devfileJson = safeYAMLToJS(values.devfile.devfileContent);
+  }
+  const applicationType =
+    (!values.application?.selectedKey && values.application?.name) ||
+    values.application?.selectedKey === values.application?.name
+      ? 'default'
+      : values.application?.selectedKey === CREATE_APPLICATION_KEY && !!values.application?.name
+      ? 'custom'
+      : 'none';
+  const resourceType =
+    values.resources === Resources.Kubernetes
+      ? 'Deployment'
+      : values.resources === Resources.OpenShift
+      ? 'DeploymentConfig'
+      : values.resources === Resources.KnativeService && 'Knative Service';
+  const selStrategy = values.import?.selectedStrategy?.name;
+  return {
+    useRecommended: values.image?.recommended === values.image?.selected,
+    recommendedStrategy: values.import?.recommendedStrategy?.name,
+    recommendedBuilderImage: values.image?.recommended,
+    strategy: selStrategy,
+    builderImage: values.image?.selected,
+
+    devFileLanguage: devfileJson?.metadata?.projectType || '',
+    devFileProjectType: devfileJson?.metadata?.language || '',
+
+    application: applicationType,
+
+    resource: resourceType,
+
+    targetPortChanged: !!values.route?.unknownTargetPort,
+
+    // currently run command is being used only for Node app
+    useRunCommand: !!values.image?.imageEnv?.NPM_RUN,
+
+    useAdvancedOptionsGit: values.git.dir !== '/' || !!values.git.ref || !!values.git.secret,
+
+    useAdvancedOptionsBuild:
+      values.build?.env.length > 0 ||
+      !values.build?.triggers?.config ||
+      !values.build?.triggers?.image ||
+      !values.build?.triggers?.webhook,
+    useAdvancedOptionsDeployment:
```

Comment:
@invincibleJai I think you meant to add `!!` and not `!` to convert these values into boolean. Or am I missing something?

```suggestion
      !!values.build?.triggers?.config ||
      !!values.build?.triggers?.image ||
      !!values.build?.triggers?.webhook,
    useAdvancedOptionsDeployment:
```

Reply:
for these default value is boolean only i.e `true` in this case. So if any of these is false then the advance deployment option would be used thus doing `!`

Reply:
Okay but in case where the user sets it to `false` and then again change it to `true` using the form in that case the build option would have been used but these values will be `false`. Wdyt?

Reply:
yeah right, current implementation checks if values has been changed if we would need to track above i.e user changes it and then changes back to the default value then would need to maintain a state in formik for it as well and keep track of these options here. 

@christianvogt let us know what should we track here change in value i.e different form default or maintain state and track for any change?